python/pyspark/sql/datasource.py

```diff
@@ -654,7 +654,7 @@ def abort(self, messages: List["WriterCommitMessage"]) -> None:
         Parameters
         ----------
         messages : list of :class:`WriterCommitMessage`\\s
-            A list of commit messages.
+            A list of commit messages. Can contain `None` values if any task failed.
```

Comment:
`List[Optional["WriterCommitMessage"]]`

I would change the type hint too.

Reply:
Actually, I will improve this in a separate PR. I think there are other type hints that need modification.